### ISSUE TITLE
field-reassign-with-default lint gives false positives because it doesn't consider if all fields are public

### ISSUE THREAD

**Issue by sdroege** [rust-lang/rust-clippy#6515]
Lint name: field-reassign-with-default

I tried this code:

```rust
let mut pretty_config = ron::ser::PrettyConfig::default();
pretty_config.new_line = "".to_string();
```

I expected to see this happen: No warning because

```rust
let pretty_config = ron::ser::PrettyConfig {
    new_line: "".to_string(),
    ..Default::default()
};
```

Doesn't compile:

```
error[E0451]: field `_future_proof` of struct `ron::ser::PrettyConfig` is private
   --> test.rs:18:15
    |
 18 |             ..Default::default()
    |               ^^^^^^^^^^^^^^^^^^ field `_future_proof` is private
```

Instead, this happened: I got a warning:

```
error: field assignment outside of initializer for an instance created with Default::default()
   --> test.rs:17:9
    |
 17 |         pretty_config.new_line = "".to_string();
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D clippy::field-reassign-with-default` implied by `-D warnings`
note: consider initializing the variable with `ron::ser::PrettyConfig { new_line: "".to_string(), ..Default::default() }` and removing relevant reassignments
```

### Meta

- `cargo clippy -V`: clippy 0.0.212 (e1884a8 2020-12-29)
- `rustc -Vv`:
```
rustc 1.49.0 (e1884a8e3 2020-12-29)
binary: rustc
commit-hash: e1884a8e3c3e813aada8254edfa120e85bf5ffca
commit-date: 2020-12-29
host: x86_64-unknown-linux-gnu
release: 1.49.0
```

**Comment by xFrednet** [comment]
This is a duplicate of #6344. The fix has been merged two days ago in #6375 🙃

**Comment by sdroege** [comment]
I guess that's too late for 1.49? :)

**Comment by xFrednet** [comment]
Looking at the [change log draft](https://github.com/rust-lang/rust-clippy/blob/b81111bc35ac0a9251d9f88804899d878a33e5a6/CHANGELOG.md) it actually seams like it will be in 1.51. The PR sadly got merged after the 1.50 cut of point if I understood it correctly 🙃

**Comment by sdroege** [comment]
No problem, I'll just disable the lint until 1.51 problem then :)

**Comment by flip1995** [comment]
Closing because it is fixed. I will create a beta backport, so the fix will get into 1.50.
